naga/src/front/wgsl/lower/mod.rs

```diff
@@ -1778,12 +1778,14 @@ impl<'source, 'temp> Lowerer<'source, 'temp> {
 
                 return Ok(());
             }
-            ast::StatementKind::Ignore(expr) => {
+            ast::StatementKind::Phony(expr, span) => {
                 let mut emitter = Emitter::default();
                 emitter.start(&ctx.function.expressions);
 
-                let _ = self.expression(expr, &mut ctx.as_expression(block, &mut emitter))?;
+                let value = self.expression(expr, &mut ctx.as_expression(block, &mut emitter))?;
                 block.extend(emitter.finish(&ctx.function.expressions));
+                ctx.named_expressions
+                    .insert(value, (format!("_naga_phony_{}", value.index()), span));
```

Comment:
I *think* if you just use the empty string for the name, everything will just work. Backends are supposed to apply the namer to everything from `naga::Function::named_expressions`, as done [here](https://github.com/gfx-rs/wgpu/blob/765c20235ea712fd14ad1b9e36bbb2071a86e319/naga/src/back/wgsl/writer.rs#L634), and that will return a fresh name for every string, including "".

Or, for legibility, just use "phony" as the name for all of them.

Please expand the snapshot test to show that these actually don't clash.

Reply:
done in https://github.com/gfx-rs/wgpu/pull/6328/commits/0036482fe5993f3ab5f47eef10a7a720f8740861